File: CS/SpringStyleItem/MyBarAndDockingController.cs
using System.ComponentModel;
using DevExpress.XtraBars;
using DevExpress.XtraBars.Styles;

namespace SpringStyleItem
{
	public class MyBarAndDockingController : BarAndDockingController
	{
		private BarItemCollection springStyleItems;

		public MyBarAndDockingController(IContainer container)
			: base(container)
		{
            Init();
        }
		public MyBarAndDockingController()
			: base()
        {
            Init();
        }

        private void Init()
        {
            springStyleItems = new BarItemCollection(this);
        }

        protected internal new void OnChanged()
		{
			base.OnChanged();
		}

        public void AddBarControlInfo()
		{
			foreach ( BarManagerPaintStyle paintStyle in this.PaintStyles )
			{
				BarControlInfo info = paintStyle.BarInfoCollection["DockedBarControl"];
				if ( info != null )
				{
					BarControlInfoCollection infos = new BarControlInfoCollection(paintStyle);
					for ( int i = 0; i < paintStyle.BarInfoCollection.Count; i++ )
					{
						BarControlInfo barInfo = paintStyle.BarInfoCollection[i];
						if ( barInfo.Name != "DockedBarControl" )
							infos.Add(new BarControlInfo(barInfo.Name, barInfo.ItemType, barInfo.ViewInfoType, barInfo.Painter));
						else
							infos.Add(new BarControlInfo(barInfo.Name, barInfo.ItemType, typeof(MyDockedBarControlViewInfo), barInfo.Painter));
					}

					paintStyle.BarInfoCollection.Clear();
					for ( int i = 0; i < infos.Count; i++ )
						paintStyle.BarInfoCollection.Add(new BarControlInfo(infos[i].Name, infos[i].ItemType, infos[i].ViewInfoType, infos[i].Painter));
				}
			}
		}

		[DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
		public BarItemCollection SpringStyleItems
		{
			get
			{
				return springStyleItems;
			}
		}
	}
}

File: VB/SpringStyleItem/MyBarAndDockingController.vb
Imports System.ComponentModel
Imports DevExpress.XtraBars
Imports DevExpress.XtraBars.Styles

Namespace SpringStyleItem
    Public Class MyBarAndDockingController
        Inherits BarAndDockingController


        Private springStyleItems_Renamed As BarItemCollection

        Public Sub New(ByVal container As IContainer)
            MyBase.New(container)
            ChangeDockedBarControlViewInfo()
        End Sub
        Public Sub New()
            MyBase.New()
            ChangeDockedBarControlViewInfo()
        End Sub

        Protected Friend Shadows Sub OnChanged()
            MyBase.OnChanged()
        End Sub

        Private Sub ChangeDockedBarControlViewInfo()
            springStyleItems_Renamed = New BarItemCollection(Me)

            For Each paintStyle As BarManagerPaintStyle In Me.PaintStyles
                Dim info As BarControlInfo = paintStyle.BarInfoCollection("DockedBarControl")
                If info IsNot Nothing Then
                    Dim infos As New BarControlInfoCollection(paintStyle)
                    For i As Integer = 0 To paintStyle.BarInfoCollection.Count - 1
                        Dim barInfo As BarControlInfo = paintStyle.BarInfoCollection(i)
                        If barInfo.Name <> "DockedBarControl" Then
                            infos.Add(New BarControlInfo(barInfo.Name, barInfo.ItemType, barInfo.ViewInfoType, barInfo.Painter))
                        Else
                            infos.Add(New BarControlInfo(barInfo.Name, barInfo.ItemType, GetType(MyDockedBarControlViewInfo), barInfo.Painter))
                        End If
                    Next i

                    paintStyle.BarInfoCollection.Clear()
                    For i As Integer = 0 To infos.Count - 1
                        paintStyle.BarInfoCollection.Add(New BarControlInfo(infos(i).Name, infos(i).ItemType, infos(i).ViewInfoType, infos(i).Painter))
                    Next i
                End If
            Next paintStyle
        End Sub

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
        Public ReadOnly Property SpringStyleItems() As BarItemCollection
            Get
                Return springStyleItems_Renamed
            End Get
        End Property
    End Class
End Namespace
